### DISCUSSION TITLE
Should able to capture line number click event.

### DISCUSSION THREAD

**Discussion by leogong99** [microsoft/monaco-editor#3641] [Q&A]
We would like to keep selected line numbers in the URL so next time user refresh the browser, all selected lines are still selected. 

Monaco-editor should have a way to capture click events and also allow to set default selected lines.

**Comment by spahnke** [comment]
> allow to set default selected lines

See this example https://microsoft.github.io/monaco-editor/playground.html#interacting-with-the-editor-revealing-a-position on how to reveal a position in the editor, that you can modify like below, to reveal and select a range in the editor.

```js
// editor.revealPositionInCenter({ lineNumber: 50, column: 120 });
const range = {startLineNumber: 50, startColumn: 10, endLineNumber: 52, endColumn: 3};
editor.revealRangeInCenter(range);
editor.setSelection(range);
```

> capture click events

See this example https://microsoft.github.io/monaco-editor/playground.html#interacting-with-the-editor-listening-to-mouse-events

**Comment by hediet** [comment]
@spahnke thanks for helping out!
